### PR TITLE
pinning hardened k8s with build-meta

### PR DESCRIPTION
**Pull request by luthermonson**

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -95,7 +95,7 @@ RUN rm -vf /charts/*.sh /charts/*.md
 # must be placed in bin/ of the file image and subdirectories of bin/ will be flattened during installation.
 # This means bin/foo/bar will become bin/bar when rke2 installs this to the host
 FROM rancher/k3s:v1.21.1-rc2-k3s1 AS k3s
-FROM rancher/hardened-kubernetes:v1.21.1 AS kubernetes
+FROM rancher/hardened-kubernetes:v1.21.2-build20210702 AS kubernetes
```
Comment (brandond) [inline Dockerfile:98]: Should look like
```suggestion
FROM rancher/hardened-kubernetes:v1.21.2-rke2r1-build20210702 AS kubernetes
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (luthermonson) [reply Dockerfile:98]: sure np

Comment (brandond) [reply Dockerfile]: Close - need to strip out and rewrite the version in the build repo:
```console
brandond@dev01:~$ docker run --rm -it rancher/hardened-kubernetes:v1.21.2-rke2r1-build20210702-linux-amd64 kube-apiserver --version
Kubernetes v1.21.2-rke2r1-build20210702
```

this should say `Kubernetes v1.21.2+rke2r1`

Comment (luthermonson) [reply Dockerfile]: oh it's passing in to the build flags and compiling it in. got it... ill fix

Comment (luthermonson) [reply Dockerfile]:
```~\r\rke2 [pin-hardened-k8s] > docker run -it rancher/hardened-kubernetes:v1.21.2-rke2r1-build20210706 kube-apiserver --version
Unable to find image 'rancher/hardened-kubernetes:v1.21.2-rke2r1-build20210706' locally
v1.21.2-rke2r1-build20210706: Pulling from rancher/hardened-kubernetes
d36999869101: Already exists
27e66ff83bb0: Already exists
fe98164e91c3: Pull complete
e09059605dea: Pull complete
cfb9e718d309: Pull complete
9abc20a9c70c: Pull complete
Digest: sha256:448f81625479014938e704cd112055c18c80f15a74a6a4e31f51f9191a5977e4
Status: Downloaded newer image for rancher/hardened-kubernetes:v1.21.2-rke2r1-build20210706
Kubernetes v1.21.2+rke2r1```

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**